music21/harmony.py

```diff
@@ -1691,7 +1709,7 @@ def typeAlter(hD):
                 degree = degree.replace('A', '') #A is for 'Altered'
                 if hD.degree == int(degree):
                     # transpose by semitones (positive for up, negative for down)
-                    p = p.transpose(hD.interval)
+                    p.transpose(hD.interval, inPlace=True),
```

Comment:
This changes the original pitch -- is this what is wanted?  Need tests to be sure.

Reply:
Previous version was a no-op, the resulting pitch was ignored and the chord modification not applied. This was caught by one of the tests I added. I can certainly rewrite it in a functional, non-inPlace mode (which seems to be in the spirit of this method return the new list of pitches).